config.go

```diff
@@ -1362,13 +1363,65 @@ func extractBitcoindRPCParams(bitcoindConfigPath string) (string, string, string
 }
 
 // checkZMQOptions ensures that the provided addresses to use as the hosts for
-// ZMQ rawblock and rawtx notifications are different.
+// ZMQ rawblock and rawtx notifications are valid and different.
 func checkZMQOptions(zmqBlockHost, zmqTxHost string) error {
-	if zmqBlockHost == zmqTxHost {
-		return errors.New("zmqpubrawblock and zmqpubrawtx must be set" +
+	// ZMQ can be used with diferent transports
+	// (http://api.zeromq.org/4-0:zmq-bind). At first, parse the socket URL
+	// and perform generalized comparison.
+	zmqBlockURL, err := url.Parse(zmqBlockHost)
+	if err != nil {
+		return err
+	}
+
+	zmqTxURL, err := url.Parse(zmqTxHost)
+	if err != nil {
+		return err
+	}
+
+	if zmqBlockURL == zmqTxURL {
+		return errors.New("zmqpubrawblock and zmqpubrawtx must be set " +
 			"to different addresses")
 	}
 
+	// TCP transport specific checks.
+	if zmqBlockURL.Scheme == "tcp" && zmqTxURL.Scheme == "tcp" {
+		zmqBlockResolved, err := net.ResolveTCPAddr("tcp", zmqBlockURL.Host)
+		if err != nil {
+			return err
+		}
+		zmqBlockHost, zmqBlockPort, err :=
+			net.SplitHostPort(zmqBlockResolved.String())
+		if err != nil {
+			return err
+		}
+		zmqBlockHostIP := net.ParseIP(zmqBlockHost)
+
+		zmqTxResolved, err := net.ResolveTCPAddr("tcp", zmqTxURL.Host)
```

Comment:
You should add some comments to what is being checked here.

Reply:
Done.